ecl/hql/hqlgram.y

```diff
@@ -7664,22 +7664,22 @@ simpleDataRow
                         }
     | HTTPorSOAPcall '(' expression ',' expression ',' expression ',' recordDef ')'
                         {
-                            //http
+                            if ($1.getInt() != (__int64) no_httpcall)
+                                parser->reportError(ERR_EXPECTED, $1, "Expected HTTPCALL rather than SOAPCALL");
                             parser->normalizeExpression($3);
                             parser->normalizeExpression($5);
                             parser->normalizeExpression($7);
-                            OwnedHqlExpr flags = parser->processHttpMarkupFlag($1.getInt());
```

Comment:
Isn't this still needed to create the jsonAtom attribute?

Reply:
@ghalliday ensuring the jsonAtom attribute is necessary when overriding the no_soapcall operator, but in this case no_httpcall is being used and in fact the request is a simple URL without content of any type.